velox/connectors/hive/tests/HivePartitionFunctionTest.cpp

```diff
@@ -35,13 +36,30 @@ class HivePartitionFunctionTest : public ::testing::Test {
     keyChannels.emplace_back(0);
     connector::hive::HivePartitionFunction partitionFunction(
         bucketCount, bucketToPartition, keyChannels);
-
     std::vector<uint32_t> partitions(size);
     partitionFunction.partition(*rowVector, partitions);
     for (auto i = 0; i < size; ++i) {
       EXPECT_EQ(expectedPartitions[i], partitions[i])
           << "at " << i << ": " << vector->toString(i);
     }
+
+    // Retry the same with nested dictionaries where indices out of range
+    // indices are bad. Ensure that out of scope indices are not accessed.
+    auto innerIndices = makeIndicesInReverse(size);
+    auto outerIndices = makeIndices(size, [](auto i) { return i; });
+    outerIndices->asMutable<vector_size_t>()[size - 1] =
+        std::numeric_limits<int32_t>().max();
+    auto dictValues = BaseVector::wrapInDictionary(
```

Comment:
nit: use helper methods from the base class that don't require passing nulls buffer or size.

```
wrapInDictionary(outerIndices, size - 1, wrapInDictionary(innerIndices, vector));
```